napari/components/viewer_model.py

```diff
@@ -118,24 +143,17 @@ class ViewerModel(KeymapProvider, MousemapProvider, EventedModel):
 
     # Using allow_mutation=False means these attributes aren't settable and don't
     # have an event emitter associated with them
-    axes: AxesOverlay = Field(
-        default_factory=AxesOverlay, allow_mutation=False
-    )
     camera: Camera = Field(default_factory=Camera, allow_mutation=False)
     cursor: Cursor = Field(default_factory=Cursor, allow_mutation=False)
     dims: Dims = Field(default_factory=Dims, allow_mutation=False)
     grid: GridCanvas = Field(default_factory=GridCanvas, allow_mutation=False)
     layers: LayerList = Field(
         default_factory=LayerList, allow_mutation=False
     )  # Need to create custom JSON encoder for layer!
-    scale_bar: ScaleBarOverlay = Field(
-        default_factory=ScaleBarOverlay, allow_mutation=False
-    )
-    text_overlay: TextOverlay = Field(
-        default_factory=TextOverlay, allow_mutation=False
+    overlays: EventedDict[str, Overlay] = Field(
```

Comment:
Is there a big advantage in making this a mutable dict vs. a fixed schema (e.g. like the previous `Overlays`)?

Reply:
I see. If you are looking to split this PR up a little, I think this could be candidate as something to do later. While I'm behind the motivation (i.e. plugins that contribute overlays, or plain scripts that add them), the details for doing so are pretty complex if I understand correctly - you'd need to add model overlay, a visual overlay (which is a private concept), modify the  mapping between the two, and maybe handling UI interaction).

If you're not looking to split this up, then feel free to ignore this comment and forge ahead.

Reply:
Definitely co.plex, and no kmmediate plans to expose it. However, what do you suggest splitting out? I don't think I could simplify a lot by simply making this immutable.

Reply:
My worry was this PR does expose some part of it `viewer.overlays`. But I guess you're saying that we're exposing a complete way to contribute a new overlay. I think that's fine and we can trust napari users to be grownups and not modify this dict unless they are prepared to deal with the consequences!

Reply:
We can (should?) make this dictionary private as well.

Reply:
I changed the overlays dict to a private attribute (just like in `Layer`); this means that no mutability is exposed yet.